### PR TITLE
Replace from file through replacement editor #1458

### PR DESCRIPTION
**Pull request by ujjwal96**
Replacement file can be specified from replacement editor. 
I think this kind of defeats the purpose of `--replace-from-file` though, because now a user can also do `--replace :~q:foo:file://<filepath>` 
Should this only be allowed in replacement editor and not from `--replace` ?

### CODE REVIEW

**Comment by cortesi** [comment]
This is  a step in the right direction. Could you please:

- Rebase this on the current options work I've just done. It's all different now, but better. 
- Use @ for the "read from file" marker. This is what I've done in proxyauth and is a convention familiar from some other tools (e.g. curl). So, the invocation would be --replace /filter/foo/@~/path/to/file. 
- Remove the replace_from_file option entirely. 
- Ignore the grid editor for now. I'm going to rewrite the options editing stuff entirely in light of the new options structures. Leave that aspect for me, and just focus on the options in core for now.

**Comment by ujjwal96** [comment]
Should I change the docs as well?

**Comment by mhils** [comment]
That'd be great, yes. Don't worry if you don't feel too comfortable in written english, one of us will read over it. :)

**Comment by mhils** [comment]
Looks great! Do we want to merge Replace and _ReplaceBase at this point?

**Comment by cortesi** [comment]
Yes, let's ditch ReplaceBase, and then this looks ready to merge.

**Comment by mhils** [comment]
Not sure why this is failing on Linux and OSX. @Kriechi, any hints?

**Comment by Kriechi** [comment]
The coverage check blocks this - only 88% for the replace addon.

**Comment by mhils** [comment]
Yes, but that's a symptom of failing tests before that: https://travis-ci.org/mitmproxy/mitmproxy/jobs/210326235#L2528

File: mitmproxy/addons/replace.py [hunk 1]
```diff
@@ -72,6 +72,10 @@ def configure(self, options, updated):
                     raise exceptions.OptionsError(
                         "Invalid regular expression: %s - %s" % (rex, str(e))
                     )
+                if s.startswith("@") and not os.path.isfile(s[1:]):
```
Comment (ujjwal96) [inline mitmproxy/addons/replace.py]: I think we should use os.path.expanduser here since we are doing that in the editor and also in the replace function.

**Pull request merged**